bgpd/bgp_nb_config.c

```diff
@@ -5459,13 +5479,99 @@ int bgp_peer_groups_peer_group_afi_safis_afi_safi_enabled_modify(
 	return NB_OK;
 }
 
+void bgp_global_afi_safis_afi_safi_unicast_network_config_apply_finish(
+	struct nb_cb_apply_finish_args *args)
+{
+	const struct lyd_node *af_dnode;
+	struct bgp *bgp;
+	const char *af_name;
+	struct prefix prefix;
+	bool is_backdoor = false;
+	uint32_t label_index;
+	const char *label_str = NULL;
+	const char *rmap_name = NULL;
+	afi_t afi;
+	safi_t safi;
+
+	bgp = nb_running_get_entry(args->dnode, NULL, true);
+
+	af_dnode = yang_dnode_get_parent(args->dnode, "afi-safi");
+	af_name = yang_dnode_get_string(af_dnode, "./afi-safi-name");
+	yang_afi_safi_identity2value(af_name, &afi, &safi);
```

Comment:
The problem with the approach of setting a context when entering `adddress-family <afi> safi> are following: 
the primary stumble point is when to unset such context from running-configuration. There is no explicit `no address-family <afi> <safi>. 
Second, user can change address-family (from one to another) without explicitly applying `exit-address-family` or `exit` command.

Reply:
> the primary stumble point is when to unset such context from running-configuration. There is no explicit no address-family <afi> <safi>.

I think Vishal had a similar problem with staticd and introduced a function called `yang_get_subtree_with_no_sibling()` to avoid memleaks of this kind. In any case, even without using that function, I think that leaking a small amount of memory per AFI/SAFI combination would be acceptable considering that that change would allow a drastic simplification in the BGP-AF CLI commands and NB callbacks.

> Second, user can change address-family (from one to another) without explicitly applying exit-address-family or exit command.

When that happens the "exit" command is called implicitly by the CLI code, so `vty->xpath` is always updated accordingly.

This is a non-blocking comment, I'll approve this PR now.